src/main/java/com/minecolonies/coremod/util/StructureWrapper.java

```diff
@@ -247,11 +249,12 @@ private void placeStructure(@NotNull final BlockPos pos, final boolean complete)
                         delayedBlocks.add(localPos);
                     }
 
-                    if (this.structure.getBlockInfo(localPos).tileentityData != null && world.getTileEntity(worldPos) instanceof TileEntityFlowerPot)
+                    if (this.structure.getBlockInfo(localPos).tileentityData != null)
                     {
-                        final TileEntityFlowerPot tileentityflowerpot = (TileEntityFlowerPot) world.getTileEntity(worldPos);
-                        tileentityflowerpot.readFromNBT(this.structure.getBlockInfo(localPos).tileentityData);
-                        world.setTileEntity(worldPos, tileentityflowerpot);
+                        final NBTTagCompound tileEntityData = this.structure.getBlockInfo(localPos).tileentityData;
```

Comment:
We should cache this. move the call out of the if as it has to be called once anyway. And then use the variable inside the if